npm_and_yarn/helpers/lib/npm/conflicting-dependency-parser.js

```diff
@@ -18,22 +18,28 @@ async function findConflictingDependencies(directory, depName, targetVersion) {
   });
 
   return await arb.loadVirtual().then((tree) => {
-    var parents = [];
-
+    const parents = [];
     for (const node of tree.inventory.query("name", depName)) {
       for (const edge of node.edgesIn) {
         if (!semver.satisfies(targetVersion, edge.spec)) {
-          var parentVersion;
-          for (const fromEdge of edge.from.edgesIn.values()) {
-            if (fromEdge.name == edge.from.name) {
-              parentVersion = edge.from.version;
+          findTopLevelEdges(edge).forEach((topLevel) => {
+            if (topLevel.to === edge.from) {
+              parents.push({
+                name: edge.from.name,
+                version: edge.from.version,
+                requirement: edge.spec,
+              });
+            } else {
+              parents.push({
+                name: topLevel.to.name,
+                version: topLevel.to.version,
+                subdependency: {
```

Comment:
Should this also include `requirement` for the top level dep so we don't break the existing error rendering? I guess we'll have a period where we'll only show the top level parent depending on the vulnerable dependency which seems fine.

Small nit, wondering about the naming `subdependency` as we don't use that anywhere in core except for spec fixtures. Should it just be `dependency`?

Reply:
> Should this also include requirement for the top level dep so we don't break the existing error rendering?

We can't use this structure as is in the existing rendering, if we would include the requirement there, it would be the requirement on the subdependency, but we don't always have that, since the edge between the top-level dependency and the immediate parent of the dependency we're trying to update could be nested deeply.

Reply:
> We can't use this structure as is in the existing rendering

Ah yes, in which case can we ship this without breaking the existing rendering?

Reply:
I think we'd have to teach the rendering to deal with this change first. I'm looking into returning a flat message instead